lib/ansible/modules/web_infrastructure/ansible_tower/tower_inventory_source_update.py

```diff
@@ -0,0 +1,142 @@
+    sample: True
+details:
+    description: The results of the source update
+    returned: success
+    type: list
+    sample: [
+        "Line 1",
+        "Line 2"
+    ]
+'''
+
+from ansible.module_utils.ansible_tower import TowerModule, tower_auth_config, tower_check_mode
+
+try:
+    import tower_cli
+
+    from tower_cli.conf import settings
+    from tower_cli.api import client
+    from tower_cli.exceptions import ServerError, ConnectionError, BadRequest, TowerCLIError
+except ImportError:
+    pass
+
+
+def main():
+    argument_spec = dict(
+        inventory=dict(required=True),
+        source=dict(required=True),
+        timeout=dict(required=False, type=int, default=300),
+    )
+
+    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    inventory = module.params.get('inventory')
+    source = module.params.get('source')
+    timeout = module.params.get('timeout')
+    result = {'inventory': inventory, 'source': source}
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        # We are going to do more than just a ping chack on this module
+        # tower_check_mode(module)
+
+        # First we will do a standard ping check
+        try:
+            ping_result = client.get('/ping').json()
+            # Stuff the version into the results as an FYI
+            result['tower_version'] = ping_result['version']
+        except(ServerError, ConnectionError, BadRequest) as excinfo:
+            result['msg'] = "Failed to reach Tower: {0}".format(excinfo)
+            module.fail_json(**result)
+
+        # Now that we know we can connect, lets verify that we can resolve the inventory
+        try:
+            inventory_object = tower_cli.get_resource("inventory").get(**{'name': inventory})
+            result['inventory_id'] = inventory_object['id']
+        except TowerCLIError as e:
+            result['msg'] = "Failed to find inventory: {0}".format(e)
+            module.fail_json(**result)
+
+        # Since we were able to find the inventory, we can now check for the inventory source
+        source_resource = tower_cli.get_resource('inventory_source')
+        try:
+            source_object = source_resource.get(name=source, inventory=inventory_object['id'])
+            result['source_id'] = source_object['id']
+        except TowerCLIError as e:
+            result['msg'] = "Failed to find inventory source: {0}".format(e)
+            module.fail_json(**result)
+
+        # if we are in check mode we can return now
+        if module.check_mode:
+            result['msg'] = "Check mode passed"
+            module.exit_json(**result)
+
+        try:
+            update_response = source_resource.update(source_object['id'], monitor=False, wait=True, timeout=timeout)
+        except Exception as e:
+            result['msg'] = 'Failed to update inventory'
+            result['exception'] = e.message
+            module.fail_json(**result)
```

Comment:
If the source is not provided, there is an endpoint that lets you update all inventory sources for an inventory

(future enhancement, I would not suggest doing that here, just pointing it out)

Reply:
Good idea, but instead of leaving the source off, I think I would want to pass in ALL as the source. That way, if someone forgets the source we don't just update everything.